File: packages/dataset_config_python/src/dataset_config_python/models/dataset.py
"""Dataset model — mirrors Inspect AI's Dataset/MemoryDataset."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel

from dataset_config_python.models.sample import Sample


class Dataset(BaseModel):
    """A named collection of samples, or a reference to a file-backed dataset.

    Supports three dataset formats:
    - ``format="memory"`` (default): inline samples via ``samples`` list.
    - ``format="json"``: loads via Inspect AI's ``json_dataset(source, **args)``.
    - ``format="csv"``: loads via Inspect AI's ``csv_dataset(source, **args)``.
    """

    samples: list[Sample] = []
    """The sample records (only used when format is 'memory')."""

    name: str = ""
    """Display name for the dataset."""

    location: str | None = None
    """Dataset location (file path or remote URL)."""

    shuffled: bool = False
    """Whether the dataset was shuffled after reading."""

    format: str = "memory"
    """Dataset format: 'memory' (inline samples), 'json', or 'csv'."""

    source: str | None = None
    """File path or URL for json/csv datasets."""

    args: dict[str, Any] | None = None
    """Extra kwargs passed to json_dataset() or csv_dataset()."""

File: packages/dataset_config_dart/lib/src/models/dataset.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'sample.dart';

part 'dataset.freezed.dart';
part 'dataset.g.dart';

/// Dart representation of Inspect AI's `Dataset` / `MemoryDataset` class.
///
/// A sequence of [Sample] objects.
///
/// This models the `MemoryDataset` variant which holds samples in an
/// in-memory list.
///
/// See [`Dataset`](https://inspect.aisi.org.uk/reference/inspect_ai.dataset.html#dataset)
/// and [`MemoryDataset`](https://inspect.aisi.org.uk/reference/inspect_ai.dataset.html#memorydataset).
@freezed
sealed class Dataset with _$Dataset {
  const factory Dataset({
    /// The list of sample objects (only used when format is 'memory').
    @Default([]) List<Sample> samples,

    /// Dataset name.
    String? name,

    /// Dataset location (file path or remote URL).
    String? location,

    /// Whether the dataset was shuffled after reading.
    @Default(false) bool shuffled,

    /// Dataset format: 'memory' (inline samples), 'json', or 'csv'.
    @Default('memory') String format,

    /// File path or URL for json/csv datasets.
    String? source,

    /// Extra kwargs passed to json_dataset() or csv_dataset().
    Map<String, dynamic>? args,
  }) = _Dataset;

  factory Dataset.fromJson(Map<String, dynamic> json) =>
      _$DatasetFromJson(json);
}
